imblearn/keras/generator.py

```diff
@@ -1,230 +0,0 @@
-"""Implement generators for ``keras`` which will balance the data."""
-from __future__ import division
-
-import pytest
-
-from sklearn.base import clone
-from sklearn.utils import safe_indexing
-from sklearn.utils import check_random_state
-from sklearn.utils.testing import set_random_state
-
-from ..under_sampling import RandomUnderSampler
-from ..utils import Substitution
-from ..utils._docstring import _random_state_docstring
-
-keras = pytest.importorskip("keras")
-
-
-class BalancedBatchGenerator(keras.utils.Sequence):
-    """Create balanced batches when training a keras model.
-
-    Create a keras ``Sequence`` which is given to ``fit_generator``. The
-    sampler defines the sampling strategy used to balance the dataset ahead of
-    creating the batch. The sampler should have an attribute
-    ``return_indices``.
-
-    Parameters
-    ----------
-    X : ndarray, shape (n_samples, n_features)
-        Original imbalanced dataset.
-
-    y : ndarray, shape (n_samples,) or (n_samples, n_classes)
-        Associated targets.
-
-    sample_weight : ndarray, shape (n_samples,)
-        Sample weight.
-
-    sampler : object or None, optional (default=None)
-        A sampler instance which has an attribute ``return_indices``.
-
-    batch_size : int, optional (default=32)
-        Number of samples per gradient update.
-
-    random_state : int, RandomState instance or None, optional (default=None)
-        Control the randomization of the algorithm
-        - If int, ``random_state`` is the seed used by the random number
-          generator;
-        - If ``RandomState`` instance, random_state is the random number
-          generator;
-        - If ``None``, the random number generator is the ``RandomState``
-          instance used by ``np.random``.
-
-    Attributes
-    ----------
-    sampler_ : object
-        The sampler used to balance the dataset.
-
-    indices_ : ndarray, shape (n_samples, n_features)
-        The indices of the samples selected during sampling.
-
-    Examples
-    --------
-    >>> from sklearn.datasets import load_iris
-    >>> iris = load_iris()
-    >>> from imblearn.datasets import make_imbalance
-    >>> X, y = make_imbalance(iris.data, iris.target, {{0: 30, 1: 50, 2: 40}})
-    >>> y = keras.utils.to_categorical(y, 3)
-    >>> import keras
-    >>> model = keras.models.Sequential()
-    >>> model.add(keras.layers.Dense(y.shape[1], input_dim=X.shape[1],
-    ...                              activation='softmax'))
-    >>> model.compile(optimizer='sgd', loss='categorical_crossentropy',
-    ...               metrics=['accuracy'])
-    >>> from imblearn.keras import BalancedBatchGenerator
-    >>> from imblearn.under_sampling import NearMiss
-    >>> training_generator = BalancedBatchGenerator(
-    ...     X, y, sampler=NearMiss(), batch_size=10, random_state=42)
-    >>> callback_history = model.fit_generator(generator=training_generator,
```

Comment:
Why did you remove this example?

Reply:
temporary stuff. I need to find a way to make tensorflow example in tensorflow and keras in keras.